File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.go
func maxProfit(prices []int) int {
    if len(prices) == 0 {
        return 0
    }

    res := 0
    minPrice := prices[0]
    for _, price := range prices {
        if price < minPrice {
            minPrice = price
        } else {
            res = max(res, price - minPrice)
        }
    }
    return res
}

File: Data Structures & Algorithms/merge-intervals/submission-2.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        n = len(intervals)
        if n < 2:
            return intervals
        intervals.sort(key=lambda x: (x[0], x[1]))
        res = []
        prev = intervals[0]
        for j in range(1, n):
            cur = intervals[j]
            if prev[1] < cur[0]:
                res.append(prev)
                prev = cur
            else:
                prev = [min(prev[0], cur[0]), max(prev[1], cur[1])]
        res.append(prev)
        return res